lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,923 @@
+    """
+    Publish an Ansible collection tarball into an Ansible Galaxy server.
+
+    :param collection_path: The path to the collection tarball to publish.
+    :param server: A native string of the Ansible Galaxy server to publish to.
+    :param key: The API key to use for authorization.
+    :param ignore_certs: Whether to ignore certificate validation when interacting with the server.
+    """
+    if not os.path.exists(collection_path):
+        raise AnsibleError("The collection path specified '%s' does not exist." % to_native(collection_path))
+    elif not tarfile.is_tarfile(collection_path):
+        raise AnsibleError("The collection path specified '%s' is not a tarball, use 'ansible-galaxy collection "
+                           "build' to create a proper release artifact." % to_native(collection_path))
+
+    display.display("Publishing collection artifact '%s' to %s" % (to_text(collection_path), server))
+
+    url = _urljoin(server, 'api', 'v2', 'collections')
+
+    data, content_type = _get_mime_data(collection_path)
+    headers = {
+        'Content-type': content_type,
+        'Content-length': len(data),
+    }
+    if key:
+        headers['Authorization'] = "Token %s" % key
+    validate_certs = not ignore_certs
+
+    try:
+        resp = json.load(open_url(url, data=data, headers=headers, method='POST', validate_certs=validate_certs))
+    except urllib_error.HTTPError as err:
+        try:
+            err_info = json.load(err)
+        except (AttributeError, ValueError):
+            err_info = {}
+
+        code = err_info.get('code', 'Unknown')
+        message = err_info.get('message', 'Unknown error returned by Galaxy server.')
+
+        raise AnsibleError("Error when publishing collection (HTTP Code: %d, Message: %s Code: %s)"
+                           % (err.code, message, code))
+
+    display.vvv("Collection has been pushed to the Galaxy server %s" % server)
+    import_uri = resp['task']
+    if wait:
+        _wait_import(import_uri, key, validate_certs)
+        display.display("Collection has been successfully published to the Galaxy server")
+    else:
+        display.display("Collection has been pushed to the Galaxy server, not waiting until import has completed "
+                        "due to --no-wait being set. Import task results can be found at %s" % import_uri)
+
+
+def install_collections(collections, output_path, servers, validate_certs, ignore_errors, no_deps, force, force_deps):
+    """
+    Install Ansible collections to the path specified.
+
+    :param collections: The collections to install, should be a list of tuples with (name, requirement, galaxy server).
+    :param output_path: The path to install the collections to.
+    :param servers: A list of Galaxy servers to query when searching for a collection.
+    :param validate_certs: Whether to validate the Galaxy server certificates.
+    :param ignore_errors: Whether to ignore any errors when installing the collection.
```

Comment:
It's unclear if i would just use --validate_certs to Validate the certs, or --validate_certs=True to validate certs.  If I just use the flags, then this an ignore_errors should remove 'Whether to...'

Reply:
It's actually missing `--ignore-certs` CLI arg that implies `validate_certs=True` according to the code I saw...

Reply:
Yep the internal API uses validate_certs but the actual cmdline call is `ignore-certs`.